spec/actions/products/create_product_spec.cr

```diff
@@ -0,0 +1,27 @@
+require "../../spec_helper"
+require "../../../src/actions/products/create_product.cr"
+
+def product_hash
+  {"name" => "Fake"}
+end
+
+describe Products::CreateProduct do
+  describe "#create" do
+    it "when creating with invalid name must raise BadRequestException" do
+        expect_raises(BadRequestException) do
+          product = Products::CreateProduct.new({"name" => ""}).create
```

Comment:
You don't need to assign a variable because this call will raise a error (line 18 too)